### ISSUE TITLE
Name Glossary

### ISSUE THREAD

**Issue by ph1p** [reactjs/de.reactjs.org#10]
This glossary is intended for frequently used words.

|English|German|Depends on context|
|--|--|--|
|Note|Hinweis|-|
|Tip|Tipp|-|
|shim|?|x|
|polyfill|Polyfill|-|
|state|State|x|
|templates|?|-|
|Setup|Setup|-|
|immutability||-|
|immutable|immutable|-|
|Properties|Properties|x|
|Props|Props|x|
|FAQ|FAQ|-|
|lifecycle methods|Lifecycle-Methoden|-|
|pure functions|pure functions|-|
|impure functions|impure functions|-|
|View-Hierarchie|-|-|
|Guide|Leitfaden|x|
|Try it on CodePen|Probier es auf CodePen aus||
|Frontend Build Pipeline|||
|out of the box|||
|Package Manager|||
|third-party packages|Pakete von Drittanbietern||
|toolchain|Werkzeugkette or Werkzeuge|x|

**Comment by miku86** [comment]
Personally, I don't use `templates` in the React context, more in the Vue context, so I would go with `Vorlagen`.

I think `polyfill` is a neologism, that can't get translated.

I always use `state`, because it's deeply coupled with React, like in `setState()`.
I translated `state` with `Zustand`, but now I think that staying with the original `state` perhaps would be better.

In general, I'm a big fan of using the originals

**Comment by ChrisB9** [comment]
i completely agree with you for the polyfill and state part but the template is a difficult one. `Vorlage` reminds me a lot of a letter or an e-mail, something where you just fill in the blanks.. it doesn't seem right in context of react

**Comment by ChrisB9** [comment]
@ph1p Setup should stay "Setup"

**Comment by ph1p** [comment]
Should we add **Guide** instead of **Leitfaden** or **Handbuch**?

**Comment by miku86** [comment]
> Should we add **Guide** instead of **Leitfaden** or **Handbuch**?

*Handbuch* klingt im alltäglichen Sprachgebrauch fremd.
Ich finde *Leitfaden* oder auch einfach *Anleitung* gut.

**Comment by JohannesKlauss** [comment]
@ph1p I'd say for **Guide** or **Leitfaden**. **Handbuch** would imply to much I think.

How should we translate **single-page apps**? As **SPA** and imply that the reader knows what that means? I can't really think of a german translation that would sound smooth _and_ make sense.

**Comment by ph1p** [comment]
First paragraph on this site https://reactjs.org/docs/glossary.html. I think we should use **Single-Page Applications** and **SPA** as it is. Sometimes there are words we also do not translate into german like **State-Management**, **Deployment**, **Server-Side Rendering (SSA)** or something else.

**Comment by ChrisB9** [comment]
i saw it on the russian translation page (translation.md): a file dedicated for this sole purpose so it doesnt get lost in issues

**Comment by ph1p** [comment]
Spanish and Brazilian also. We should add that too. At the beginning I wasn't sure what we can add, change or delete in this repository.

**Comment by do-wa** [comment]
I think this is the right ticket for this kind of questions, because I'm facing some difficulties in terms of translating some react specific words:
**root ReactComponent** 
Ursprungs ReactComponent ?
oberste ReactComponent ?
I think the literal translation "Wurzel" isn't helpful. 

- hydrate
I saw that the verb render is translated by others in "denglish" with "gerendet" , "rendert" etc.
Is there any alternative to "hydraten" or a completely different word?

**Comment by JohannesKlauss** [comment]
I'd say Root ReactComponent: React Wurzelkomponente

**Comment by ph1p** [comment]
> I'd say Root ReactComponent: React Wurzelkomponente
> The image of an tree as in component tree is also present in german and a lot of IT stuff in german uses words like Wurzel, Blatt, Ast, Abzweigung to describe trees.


Maybe **Ursprungs-Komponente**?

**Comment by JohannesKlauss** [comment]
I don't think **Ursprungs-Komponente** is correct, since it implies it's the component all other components are derived from, which is not the case.

**Comment by pavethiran** [comment]
What could be the correct way to translate `mount`, `mounted`, `unmount` and `unmounted`?

Reference:
https://github.com/reactjs/de.reactjs.org/blame/master/content/docs/reference-react-component.md#L148

**Comment by ph1p** [comment]
> What could be the correct way to translate `mount`, `mounted`, `unmount` and `unmounted`?
> 
> Reference:
> https://github.com/reactjs/de.reactjs.org/blame/master/content/docs/reference-react-component.md#L148

Hi! I think we should use `gemounted` and `geunmounted`. A little bit of denglisch. Because it's important to keep the context with all lifecycle methods. I'm assuming no one can do anything with "montieren/befestigen/abmontieren"

**Comment by ameetmadan** [comment]
How about `props`and `state`? I have left them in english until now considering they are not necessarily needed to be translated as the code on the pages is in English as well.
